fairlearn/metrics/_group_feature.py

```diff
@@ -65,27 +63,22 @@ def __init__(self,
                 if isinstance(feature_vector.name, str):
                     self._name = feature_vector.name
                 else:
-                    msg = _SERIES_NAME_NOT_STRING.format(feature_vector.name,
-                                                         type(feature_vector.name))
+                    msg = _SERIES_NAME_NOT_STRING.format(
+                        feature_vector.name, type(feature_vector.name)
+                    )
                     raise ValueError(msg)
 
-    def get_mask_for_class(self,
-                           target_class: Any) -> np.ndarray:
-        """Fetch a mask array for the given class."""
-        idx = self.classes.index(target_class)
-        return self.get_mask_for_class_index(idx)
-
-    def get_mask_for_class_index(self,
-                                 target_class_index: int) -> np.ndarray:
-        """Fetch a mask array for the given class index."""
-        return (self._encoded == target_class_index)
-
     @property
-    def name(self) -> str:
+    def name_(self) -> str:
```

Comment:
why not store the values in these attributes and then remove the 3 properties?

Reply:
I can do that - I thought this was the SciKit-Learn convention, though? Or have I misunderstood it....

Reply:
our convention there is not to have many properties. If the user is to have access to a `name_` attribute, it's stored under `obj.name_` w/o any property to hide the _private_ attribute. We just store it publicly in the first place.

Reply:
OK. Changed to match this